### PR TITLE
Delete new branches pushed to the main repository

### PR DESCRIPTION
**Pull request by prince-chrismc**
Just to enforce everyone uses a fork.

### CODE REVIEW

**Review by sbtaylor15 — COMMENTED** [review]
New PRs are not going to trigger this to run, correct?

**Comment by prince-chrismc** [comment]
> New PRs are not going to trigger this to run, correct?

Thats the intention! PRs can include forks, there's no need to run on that. Only pushes can create local branch so that why I imposed this limitation

**Review by sbtaylor15 — APPROVED** [review]
No changes needed.

**Review by MitaliBo — APPROVED** [review]

**Comment by prince-chrismc** [comment]
I'll merge this after the daily so everyone can be warned =)

File: .github/workflows/delete-pushed-branches.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  push:
+   branches-ignore: [main]
+   
+jobs:
+  delete:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'pyrsia' # Only on main repo
+    steps:
+    - run: |
+        curl -s -X DELETE -u ${{ github.token }} https://api.github.com/repos/${{ github.repository }}/git/refs/heads/${{ github.ref }}
```
Comment (chb0github) [inline .github/workflows/delete-pushed-branches.yml:13]: mmm is this right? because with curl, `-u` has to be basic auth: `base64 <<< "username:password"`

**Review by chb0github — COMMENTED** [review]
Just one question

File: .github/workflows/delete-pushed-branches.yml (hunk 1 above)
Comment (prince-chrismc) [reply .github/workflows/delete-pushed-branches.yml:13]: I stole this from the GitHub documentation https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api#authentication

**Pull request merged**